package.json

```diff
@@ -30,7 +30,9 @@
     "moduleNameMapper": {
       "^js(.*)$": "<rootDir>/src/js$1",
       "^dojo(.*)$": "<rootDir>/__mocks__/arcgis.js",
-      "^esri(.*)$": "<rootDir>/__mocks__/arcgis.js",
+      "esri/request": "<rootDir>/__mocks__/MapView.js",
+      "esri/views/MapView": "<rootDir>/__mocks__/MapView.js",
+      "esri/Map": "<rootDir>/__mocks__/Map.js",
       "^dijit(.*)$": "<rootDir>/__mocks__/arcgis.js",
```

Comment:
I tried a couple work-arounds, I'll keep at it. 
Per the Mocks for common things, I like it.  You can also write your Mock or Spy in your beforeEach's within each test. Or, you can have something like a global beforeEach that sets up anything we'll need across components.